### PR TITLE
feat: Adding the Terraform module

### PR DESCRIPTION
**Pull request by gatici**
# Description

This PR adds the Terraform module for the charm. Hence, the charm could be deployed using Terraform.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: terraform/LICENSE [hunk 1]
```diff
@@ -0,0 +1,202 @@
+
+                                 Apache License
```
Comment (ghislainbourgeois) [inline terraform/LICENSE:2]: Do we really need the duplicate `LICENSE` file?

File: terraform/variables.tf
Comment (ghislainbourgeois) [inline terraform/variables.tf:1]: How do we pass the juju configuration for this operator? There are some optional configs in `config.yaml`.

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: terraform/LICENSE (hunk 1 above)
Comment (gatici) [reply terraform/LICENSE:2]: I was removing the LICENSE file but missed it in here. Good catch, thanks.

Comment (gatici) [reply terraform/LICENSE]: Done

File: terraform/variables.tf
Comment (gatici) [reply terraform/variables.tf:1]: The config options could be passed in main.tf while defining application. 
Here is an example applications which has config options.
```
resource "juju_application" "traefik-k8s" {
  name  = "traefik-k8s"
  model = var.model_name

  charm {
    name    = "traefik-k8s"
    channel = var.channel
  }

  config = {
    routing_mode      = "subdomain"
    external_hostname = var.external_hostname
  }
  units = 1
  trust = true

}
```

Should a Terraform module include all the config options of charm ?

Comment (markbeierl) [reply terraform/variables.tf:1]: I think we can just allow for arbitrary config to be passed through.  Thinking of all the options for the UPF, as an example, would present a large maintenance burden to keep them all in sync

Comment (gatici) [reply terraform/variables.tf:1]: The config options are added as an config dict which will be filled up by the users.

File: terraform/terraform.tfvars [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+certs_application_name = "put your self-signed-certificates app name here"
+nrf_application_name   = "put your nrf app name here"
+
+# Optional Configuration
+channel = "put the charm channel here"
+amf-config = {
+  dnn                   = "put your dnn here"
+  external-amf-ip       = "put your external AMF IP here"
+  external-amf-hostname = "put your external AMF hostname here"
+}
```
Comment (ghislainbourgeois) [inline terraform/terraform.tfvars:13]: As they are optional, how can we use the default values? Do we need to remove those lines?

File: terraform/variables.tf
Comment (ghislainbourgeois) [reply terraform/variables.tf:1]: I added a quick question regarding the default values, resolving this comment.

File: terraform/terraform.tfvars (hunk 2 above)
Comment (gatici) [reply terraform/terraform.tfvars:13]: I have commented the optional config section in the final patch.

Comment (ghislainbourgeois) [reply terraform/terraform.tfvars]: Thanks!

**Pull request merged**